### PR TITLE
[flang][OpenMP] Don't check unlabelled `cycle` branching for target loops

### PR DESCRIPTION
**Pull request by ergawy**
Properly handles `cycle` branching inside target distribute loops.

### CODE REVIEW

**Review by NimishMishra — APPROVED** [review]
Looks good. Thanks

File: flang/lib/Semantics/check-directive-structure.h [hunk 1]
```diff
@@ -74,6 +74,9 @@ template <typename D> class NoBranchingEnforce {
         case llvm::omp::Directive::OMPD_distribute_parallel_for:
         case llvm::omp::Directive::OMPD_distribute_simd:
         case llvm::omp::Directive::OMPD_distribute_parallel_for_simd:
+        case llvm::omp::Directive::OMPD_target_teams_distribute_parallel_do:
+        case llvm::omp::Directive::
+            OMPD_target_teams_distribute_parallel_do_simd:
```
Comment (mjklemm) [inline flang/lib/Semantics/check-directive-structure.h:79]: Stray formatting change?

**Review by mjklemm — APPROVED** [review]
LGTM

File: flang/lib/Semantics/check-directive-structure.h (hunk 1 above)
Comment (ergawy) [reply flang/lib/Semantics/check-directive-structure.h:79]: Nope, our formatting rules demand this.

**Pull request merged**